File: rb/Irrgarten/FuzzyPlayer.rb
#encoding:utf-8
require_relative 'Player'
module Irrgarten
    class FuzzyPlayer < Player
        def initialize(other)
            copy(other)
        end

        def move(direction, valid_moves)
            Dice.next_step(super(direction,valid_moves),valid_moves, @intelligence)
        end

        def attack
            Dice.intensity(@intelligence+sum_weapon)
        end

        def to_s
            "Fuzzy"+super
        end

        protected

        def defensive_energy
            Dice.intensity(@intelligence+sum_shield)
        end
    end
end

File: ExamenP5/Irrgarten/src/irrgarten/FuzzyPlayer.java

package irrgarten;

import java.util.ArrayList;

/**
 *
 * @author Sergio Salvador Gil
 */
public class FuzzyPlayer extends Player{
    public FuzzyPlayer (Player other){
        super(other);
    }
    
    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves){
        return (Dice.nextStep(super.move(direction, validMoves), validMoves, this.getIntelligence()));
    }
    @Override
    public float attack(){
        return (Dice.intensity(this.getStrength())+this.sumWeapon());
    }
    
    @Override
    protected float defensiveEnergy(){
        return (Dice.intensity(this.getIntelligence())+this.sumShield());
    }
    
    @Override
    public String toString(){
        return "Fuzzy"+super.toString();
    }
}
